teps/000X-trigger-crd.md

```diff
@@ -0,0 +1,351 @@
+  - [User Stories (optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Notes/Constraints/Caveats (optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+<!-- /toc -->
+
+## Summary
+
+<!--
+This section is incredibly important for producing high quality user-focused
+documentation such as release notes or a development roadmap.  It should be
+possible to collect this information before implementation begins in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+This proposal is to extract out Trigger definition from EventListener Spec into its own CRD 
+to resolve the problem of Multitenant EventListener.
+
+This is a TEP for the implementation of the Trigger CRD that were
+discussed in the following docs: 
+
+- [Trigger Listener CRDs](https://docs.google.com/document/d/1zUVrIbGZh2R9dawKQ9Hm1Cx3GevKIfOcRO3fFLdmBDc/edit#heading=h.tp9mko2koenr)
+- [Multitenant EventListener](https://docs.google.com/document/d/1NX0ExhPad6ixTM8AdU0b6Vc3MVD5hQ_vIrOs9dIXq-I/edit) 
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this TEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a TEP within the wider Tekton community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+Today, EventListener commonly are created for every namespace and handle TriggerBinding and
+TriggerTemplate in that same namespace. Creation of EventListener causes pod proliferation. For
+every namespace that requires handling of webhook, we need EventListener which in turns lead to
+this pod proliferation and causes excess resource consumption.
+
+### Goals
+
+<!--
+List the specific goals of the TEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+1. Reduction in resource cosumption due to EventListener.
+2. Ability to have single EventListener to cater to whole cluster.
```

Comment:
could you expand a bit on the motivation and goals? in my opinion these are one of the most important sections in the TEP because they help the reader understand the problem being solved fully. it's hard to judge a proposal if you don't fully understand why the thing is being proposed.

Maybe you could elaborate on each of the points you already have, e.g.:

> Today, EventListener commonly are created for every namespace and handle TriggerBinding and TriggerTemplate in that same namespace.
> Creation of EventListener causes pod proliferation. 

Can you explain a bit more, maybe with some examples? e.g. how many namespaces you are dealing with, how many pods.

> For every namespace that requires handling of webhook, we need EventListener which in turns lead to
this pod proliferation and causes excess resource consumption.

Does 'handling a webhook' here mean instantiating resources within that namespace?

Can you elaborate on "resource consumption"?